snuba/manual_jobs/runner.py

```diff
@@ -34,7 +63,69 @@ def _build_job_spec_from_entry(content: Any) -> JobSpec:
     return job_spec
 
 
+def _job_lock_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:lock"
+
+
+def _job_status_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:execution_status"
+
+
+def _acquire_job_lock(job_id: str) -> bool:
+    return bool(
+        _redis_client.set(
+            name=_job_lock_key(job_id), value=1, nx=True, ex=(24 * 60 * 60)
+        )
+    )
+
+
+def _release_job_lock(job_id: str) -> None:
+    _redis_client.delete(_job_lock_key(job_id))
+
+
+def _set_job_status(job_id: str, status: JobStatus) -> JobStatus:
+    _redis_client.set(name=_job_status_key(job_id), value=status.value)
+    return status
+
+
+def get_job_status(job_id: str) -> Optional[JobStatus]:
+    redis_status = _redis_client.get(name=_job_status_key(job_id))
+    if redis_status is None:
+        return redis_status
+    else:
+        return JobStatus(redis_status.decode("utf-8"))
+
+
 def list_job_specs(
-    jobs_filename: str = "job_manifest.json",
+    manifest_filename: str = MANIFEST_FILENAME,
 ) -> Mapping[str, JobSpec]:
-    return _read_from_path(jobs_filename)
+    return _read_manifest_from_path(manifest_filename)
+
+
+def run_job(job_spec: JobSpec, dry_run: bool) -> JobStatus:
+    current_job_status = get_job_status(job_spec.job_id)
+    if current_job_status is not None and current_job_status != JobStatus.NOT_STARTED:
+        raise JobStatusException(job_id=job_spec.job_id, status=current_job_status)
+
+    have_lock = _acquire_job_lock(job_spec.job_id)
+    if not have_lock:
+        raise JobLockedException(job_spec.job_id)
+
+    current_job_status = _set_job_status(job_spec.job_id, JobStatus.NOT_STARTED)
+
+    job_to_run = _JobLoader.get_job_instance(job_spec, dry_run)
+
+    try:
+        if not dry_run:
```

Comment:
I'm starting to feel like `dry_run` may not be worth the effort to support. I think it won't be relevant in a lot of situations and it generates a fair amount of branching

Reply:
sgtm

Reply:
Are we going to do the `dry_run` cleanup in this PR or in a follow-up PR? I don't think it's very useful, tbh. Maybe we can add it later if folks really need it. Thoughts?

Reply:
I'd rather isolate it in a follow-up PR, since removing dry_run will touch a few things not related to status tracking